src/modal/modal.scss

```diff
@@ -67,11 +78,52 @@
         margin-right: $iui-s;
       }
     }
+
+    .iui-modal-content {
+      flex-grow: 2;
+      margin: 0 -#{$iui-m};
+      padding: 0 $iui-m;
+      overflow-y: auto;
+      @supports (overflow-y: overlay) {
+        overflow-y: overlay;
+      }
+    }
   }
+}
 
-  &.iui-modal-visible {
-    visibility: visible;
-    opacity: 1;
-    transition-delay: 0s;
+@mixin iui-modal-full-page {
+  > .iui-modal-dialog {
+    display: flex;
+    flex-direction: column;
+    height: 100vh;
+    width: 100vw;
+    left: 0;
+    top: 0;
+    transform: none;
+    max-width: initial;
+    min-width: initial;
+    border-radius: 0;
+
+    &.iui-modal-animation-enter {
```

Comment:
So it should work in React now? @elephantcatdog @mayank99

Reply:
Yeah, but we only spent ~30 mins on it. There might still be some CSS changes needed, so let's wait for the React PR (at least draft).

Reply:
You mean we shouldn't merge this PR until React one is created?

Reply:
Correct.

Reply:
React PR: https://github.com/iTwin/iTwinUI-react/pull/588